### DISCUSSION TITLE
Should we rename `SablierV2LockupPro`?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#371] [Ideas]
I know we've beaten this topic to death, but I keep getting the same feedback in all Sablier V2 demos I make: that the name of the `SablierV2LockupPro` contract (and the associated labels in the frontend) is confusing.

In business, "pro" is typically associated with a premium product with a cost. But that is not the case here; quite the contrary! `SablierV2LockupPro` is an open, permissionless smart contract that can create non-linear streams for anyone in the world with an Internet connection and an Ethereum wallet.

What would you guys say about the following alternatives (I have sourced some of these suggestions from ChatGPT):

- `LockupNonLinear`
- `LockupCurve`
- `LockupCustom`
- `LockupDynamic`
- `LockupFlex`

I don't like any of these better than `Pro`, TBH. It should be fine to keep using `Pro` for the contracts as long as only smart contract devs know about this term. But what we should do, at a minimum, is to abstract away the word "pro" in the frontend.

**Comment by maxdesalle** [comment]
> I don't like any of these better than Pro, TBH. It should be fine to keep using Pro for the contracts as long as only smart contract devs know about this term. But what we should do, at a minimum, is to abstract away the word "pro" in the frontend.

I disagree. The advantage of "Pro" IMO was precisely to use it publicly for our users. If we remove it in the frontend to not confuse users, then there is no reason to leave it on our contracts and confuse developers. And I agree with those giving feedback that the "pro" naming convention might be confusing.

I like `LockupDynamic`, simply because we could then rename "pro streams" to "dynamic streams", and it sounds much better than "non linear streams". And intuitively, a "dynamic stream" makes sense: the stream is "dynamic"!

**Reply by PaulRBerg** [reply]
Good points. It'd also be quite confusing (as a standalone matter, regardless of the names under discussion) to have two naming systems; one in the contracts, and another used in the frontend exclusively.

`LockupDynamic` has also grown on me in the meantime!

**Comment by gavriliumircea** [comment]
IMO the next best things after pro are LockupCustom and LockupNonLinear, but each one has issues. The problem for LockupCustom is that we do not give the option to the user to create a stream with any rules he want, at least not yet. The problem with LockupNonLinear is that the cliff streams are not linear but they are not covered in this section. IMO the pro streams are the streams with more advanced configuration so maybe we should take into consideration something like LockupAdvanced.

**Reply by PaulRBerg** [reply]
Thanks, @gavriliumircea.

> we do not give the option to the user to create a stream with any rules he want

I think you are referring to the frontend here? Because at the contract level, the user can choose pretty much whatever mathematical function he wants.

> The problem with `LockupNonLinear` is that the cliff streams are not linear but they are not covered in this section

I think you meant to say "not non-linear"? Assuming yes, then this is a good point.

> we should take into consideration something like `LockupAdvanced`

That is an option worth considering, too.

**Comment by razgraf** [comment]
I like "Dynamic". Next choice would be "Advanced".

"Flex" sounds like we're flexing on someone 😅. 
"Non-Linear" is way too long. 
"Custom" I'd keep it for when/if we do a stream builder in the UI. As in (1) suggested-dynamic-streams (pick from the list) or (2) custom-dynamic-streams (build your own).
I liked "Curve" too but it sounds weird to say "Curve streams" in phrases like to "Linear or Curve streams".

**Reply by PaulRBerg** [reply]
Great, we all like "Dynamic". and you, @andreivladbrg?

> "Flex" sounds like we're flexing on someone

I think it's just you who is associating the word "flex" with the act of flaunting one's abilities. "Flex" is often used in business names, e.g. [FlexiSpot](https://www.flexispot.com/).

> "Custom" I'd keep it for when/if we do a stream builder in the U

SGTM

> I liked "Curve" too but it sounds weird to say "Curve streams" in phrases like to "Linear or Curve streams".

Yes. And another issue with "curve" is that it is not fit for purpose. There can be pro streams that are not strictly speaking curves, e.g. the traditional unlocks.

**Comment by andreivladbrg** [comment]
I agree with @maxdesalle that having different names between the contracts and frontend app would be confusing for developers.

Regarding the suggested options, I like "Custom" and "Dynamic". The problem with "Custom"  is that there isn't a stream builder in the UI. On the other hand, "Dynamic" may give users the impression that they can adjust/change their streams, which is not the case.

 If I had to choose one of these options, I would go with "Dynamic" because it is more catchy.

**Comment by PaulRBerg** [comment]
Going to close this since the finalized version of the repo (ready for audit) uses `SablierV2LockupDynamic`.
